forward/utils.go

```diff
@@ -17,9 +17,27 @@ func cloneURL(i *url.URL) *url.URL {
 // copyHeaders copies http headers from source to destination.  It does not
 // overide, but adds multiple headers
 func copyHeaders(dst, src http.Header) {
+	var extraHopByHopHeaders []string
 	for k, vv := range src {
-		for _, v := range vv {
-			dst.Add(k, v)
+		switch k {
```

Comment:
Great! Thank you for looking it up in Go's code.
I actually think that doesn't matter, writing all cases comma-separated is just syntactic sugar. I think probably makes sense encapsulating this as a "rewriter" since the code was ready for it, as Oxy is doing. After all, the architecture is fully based on Oxy and Go's middleware. I agree on the other hand the implementation details we are talking about here aren't that important with just a few tests.
I'll merge this and see if it makes sense to move it to the rewriter or just remove the rewriter.
Thanks for the discussion.